src/net/ssl-context.hpp

```diff
@@ -0,0 +1,29 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+#ifndef SRC_NET_SSL_CONTEXT_HPP
+#define SRC_NET_SSL_CONTEXT_HPP
+
+// Forward declarations
+struct ssl_st;
+struct ssl_ctx_st;
+
+namespace mk {
+namespace net {
+
+class SslContext {
```

Comment:
this should be not movable and not copyable.
we don't want to have copy of the singleton